### PR TITLE
A bunch of small optimisations

### PR DESCRIPTION
**Pull request by klochowicz**
See commit details.

### CODE REVIEW

File: crates/model/src/lib.rs [hunk 1]
```diff
@@ -240,8 +239,7 @@ impl Div<Price> for Contracts {
     fn div(self, rhs: Price) -> Self::Output {
         let mut btc = self.0 / rhs.0;
         btc.rescale(8);
-        Amount::from_str_in(&btc.to_string(), Denomination::Bitcoin)
-            .expect("Error computing BTC amount")
+        Amount::from_btc(btc.to_f64().expect("to fit in f64")).expect("Parse the BTC amount")
```
Comment (bonomat) [inline crates/model/src/lib.rs:242]: This does the same thing underthehood.

**Review by bonomat — APPROVED** [review]

File: crates/model/src/lib.rs (hunk 1 above)
Comment (klochowicz) [reply crates/model/src/lib.rs:242]: not quite - we avoid using `Decimal`s `Display` implementation, which is AFAICT not as straightforward as f64 (regular float).

Comment (klochowicz) [reply crates/model/src/lib.rs:242]: but good point that `Amount::from_btc` creates a `String`. I have overlooked that.

**Pull request merged**